nautilus_trader/persistence/catalog/parquet.py

```diff
@@ -134,6 +148,7 @@ def _query(  # noqa (too complex)
                 table=table, mappings=mappings, raise_on_empty=raise_on_empty, **kwargs
             )
         else:
+            table = pa.Table.from_pandas(table)
```

Comment:
Although _handle_table_nautilus should accept the dataframe, however when passing dataframe directly following error occurs. If that is fixed then there is no need for this line.
```shell
currency_pair.pyx", line 201, in nautilus_trader.model.instruments.currency_pair.CurrencyPair.from_dict_c
            TypeError: Expected unicode, got float
```